runtime/lua/vim/lsp.lua

```diff
@@ -651,7 +673,18 @@ function lsp.start(config, opts)
 
   if not config.root_dir and opts._root_markers then
     config = vim.deepcopy(config)
-    config.root_dir = vim.fs.root(bufnr, opts._root_markers)
+
+    if type(opts._root_markers) == 'table' then
+      for _, marker in ipairs(opts._root_markers) do
+        local root = vim.fs.root(bufnr, marker)
+        if root ~= nil then
+          config.root_dir = root
+          break
+        end
+      end
+    else
+      config.root_dir = vim.fs.root(bufnr, opts._root_markers)
```

Comment:
could eliminate this `else` by changing the above to:

```lua
for _, marker in ipairs(type(opts._root_markers[1] == 'string' and { opts._root_markers } or opts._root_markers) do
  ...
```

Reply:
I don't think we need the `else` at all.